cli/src/cmd/forge/verify/mod.rs

```diff
@@ -167,7 +167,9 @@ impl VerifyArgs {
 
     /// Returns the configured verification provider
     pub fn verification_provider(&self) -> eyre::Result<Box<dyn VerificationProvider>> {
-        self.verifier.verifier.client(&self.etherscan.key)
+        let config = self.load_config_emit_warnings();
```

Comment:
right, this looks suspicious

@Evalir mind giving this a closer look?

do we need to emit warnings here again? iirc we load it in the command anyways?

Reply:
So, I gave this a closer look, and I believe this is wrong and duplicating code—we already load the config on the verify command and _explicitly_ set the etherscan API key. This would cause the config to be loaded again and warnings to be duplicated.

I also don't understand why this is modifying the `verify-contract` command if the mentioned issue is on create?

Reply:
I think that this logic should be stored in verify command, because create should not perform verify-related logic of choosing etherscan keys to use. The problem, if we look at it from the different angle, is that if we call VerifyArgs.verification_provider(), it will return the provider with different etherscan_key than the one which will actually be used in VerifyArgs.run() for the same configuration. So the problem is about VerifyArgs.verification_provider() function and it seems that it is create-related just because create is the only place where it is being called

Regarding the config reading, will it be better to just call `self.load_config()` in this context?